src/thread_local_heap.h

```diff
@@ -146,7 +147,8 @@ class ThreadLocalHeap {
     }
   }
 
-  inline void *ATTRIBUTE_ALWAYS_INLINE calloc(size_t count, size_t size) {
+  inline void *ATTRIBUTE_ALWAYS_INLINE ATTRIBUTE_MALLOC ATTRIBUTE_ALLOC_SIZE2(2, 3)
```

Comment:
similarly, shouldn't this be `(1, 2)`?

Reply:
I was also confused but merged this since it turns out that for C++ non static member functions you add 1 … “The function parameter(s) denoting the allocated size are specified by one or two integer arguments supplied to the attribute. The allocated size is either the value of the single function argument specified or the product of the two function arguments specified. Argument numbering starts at one for ordinary functions, and at two for C++ non-static member functions.” (https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html)

Reply:
lol I didn't read closely enough I guess.  Okay - I get the extra offset _here_, but shouldn't these decorators be on e.g. [`mesh_realloc`](https://github.com/plasma-umass/Mesh/blob/master/src/libmesh.cc#L129) and [`mesh_calloc`](https://github.com/plasma-umass/Mesh/blob/master/src/libmesh.cc#L161)?  These C++ non-static member functions are _only_ called from those funcs in libmesh.cc